From 01b5fd69bc928370c84f4b671112937f00316cc0 Mon Sep 17 00:00:00 2001
From: Weiguang Kong <weiguang.kong@nxp.com>
Date: Wed, 18 Apr 2018 16:54:57 +0800
Subject: [PATCH 3668/5242] MLK-18073: ASoC: fsl_dsp: fix uninitialized scalar
 variable issue

commit  21d77005e722f07909eed72b1b0eab7942c4874c from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch is used to fix Coverity-1793874, Coverity-1793875,
Coverity-1793876, Coverity-1793877 issue.

The icm_base_info_t is not initialized before using, so
use memset() to initialize it to fix these issues.

Signed-off-by: Weiguang Kong <weiguang.kong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_hifi4.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sound/soc/fsl/fsl_hifi4.c b/sound/soc/fsl/fsl_hifi4.c
index 2328b7a..a36a450 100644
--- a/sound/soc/fsl/fsl_hifi4.c
+++ b/sound/soc/fsl/fsl_hifi4.c
@@ -817,6 +817,7 @@ static long fsl_hifi4_init_codec(struct fsl_hifi4 *hifi4_priv,
 	ext_msg.phys = hifi4_priv->msg_buf_phys;
 	ext_msg.size = sizeof(struct icm_base_info_t);
 
+	memset(&icm_base_info_t, 0, sizeof(struct icm_base_info_t));
 	icm_base_info_t.process_id = id;
 	icm_base_info_t.codec_id = process_info->codec_id;
 
@@ -1311,6 +1312,7 @@ static long fsl_hifi4_codec_open(struct fsl_hifi4 *hifi4_priv,
 	ext_msg.phys = hifi4_priv->msg_buf_phys;
 	ext_msg.size = sizeof(struct icm_base_info_t);
 
+	memset(&icm_base_info_t, 0, sizeof(struct icm_base_info_t));
 	icm_base_info_t.process_id = id;
 
 	memcpy(hifi4_priv->msg_buf_virt, &icm_base_info_t,
@@ -1355,6 +1357,7 @@ static int fsl_hifi4_codec_close(struct fsl_hifi4 *hifi4_priv,
 	ext_msg.phys = hifi4_priv->msg_buf_phys;
 	ext_msg.size = sizeof(struct icm_base_info_t);
 
+	memset(&icm_base_info_t, 0, sizeof(struct icm_base_info_t));
 	icm_base_info_t.process_id = id;
 
 	memcpy(hifi4_priv->msg_buf_virt, &icm_base_info_t,
@@ -1402,6 +1405,7 @@ static int fsl_hifi4_codec_reset(struct fsl_hifi4 *hifi4_priv,
 	ext_msg.phys = hifi4_priv->msg_buf_phys;
 	ext_msg.size = sizeof(struct icm_base_info_t);
 
+	memset(&icm_base_info_t, 0, sizeof(struct icm_base_info_t));
 	icm_base_info_t.process_id = id;
 
 	memcpy(hifi4_priv->msg_buf_virt, &icm_base_info_t,
-- 
1.7.9.5

